src/bp-core/bp-core-functions.php

```diff
@@ -7721,3 +7740,140 @@ function bb_validate_gravatar( $email ) {
+		'/linux/i'              => 'Linux',
+		'/ubuntu/i'             => 'Ubuntu',
+		'/iphone/i'             => 'iPhone',
+		'/ipod/i'               => 'iPod',
+		'/ipad/i'               => 'iPad',
+		'/android/i'            => 'Android',
+		'/blackberry/i'         => 'BlackBerry',
+		'/webos/i'              => 'Mobile',
+	);
+
+	foreach ( $os_array as $regex => $value ) {
+		if ( preg_match( $regex, $user_agent ) ) {
+			$os_platform = $value;
+		}
+	}
+
+	switch ( $os_platform ) {
+		case 'Windows 10':
+		case 'Windows 8.1':
+		case 'Windows 8':
+		case 'Windows 7':
+		case 'Windows Vista':
+		case 'Windows Server 2003/XP x64':
+		case 'Windows XP':
+		case 'Windows 2000':
+		case 'Windows ME':
+		case 'Windows 98':
+		case 'Windows 3.11':
+		case 'Windows 95':
+			$os_platform = 'window';
+			break;
+		case 'Mac OS X':
+		case 'Mac OS 9':
+			$os_platform = 'mac';
+			break;
+		case 'Linux':
+		case 'Ubuntu':
+			$os_platform = 'ubuntu';
+			break;
+		case 'iPhone':
+		case 'iPod':
+		case 'iPad':
+			$os_platform = 'ios_device';
+			break;
+		case 'Android':
+			$os_platform = 'android_device';
+			break;
+		case 'BlackBerry':
+			$os_platform = 'BlackBerry';
+			break;
+		case 'Mobile':
+			$os_platform = 'Mobile';
+			break;
+		default:
+			$os_platform = 'window';
+
+			break;
+	}
+
+	return $os_platform;
+}
+
+/**
+ * Get week start date with an integer Unix timestamp.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param string $start_day_of_week Week start day like monday this week.
+ *
+ * @return int
+ */
+function bb_get_week_start_timestamp( $start_day_of_week = false ) {
+	if ( empty( $start_day_of_week ) ) {
+		$start_day_of_week = 'monday this week';
+	}
+
+	$start_week      = strtotime( $start_day_of_week );
+	$start_week_date = date_i18n( 'Y-m-d', $start_week );
+	$start_week_date = $start_week_date . ' 00:00:00';
+
+	$time_chunks = explode( ':', str_replace( ' ', ':', $start_week_date ) );
+	$date_chunks = explode( '-', str_replace( ' ', '-', $start_week_date ) );
+
+	return gmmktime( (int) $time_chunks[1], (int) $time_chunks[2], (int) $time_chunks[3], (int) $date_chunks[1], (int) $date_chunks[2], (int) $date_chunks[0] );
+}
+
+/**
+ * Get week end date with an integer Unix timestamp.
+ *
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param string $end_day_of_week Week end day like sunday this week.
+ *
+ * @return int
+ */
+function bb_get_week_end_timestamp( $end_day_of_week = false ) {
+	if ( empty( $start_day_of_week ) ) {
+		$end_day_of_week = 'sunday this week';
+	}
+
+	$end_week      = strtotime( $end_day_of_week );
+	$end_week_date = date_i18n( 'Y-m-d', $end_week );
+	$end_week_date = $end_week_date . ' 23:59:59';
+
+	$time_chunks = explode( ':', str_replace( ' ', ':', $end_week_date ) );
+	$date_chunks = explode( '-', str_replace( ' ', '-', $end_week_date ) );
+
+	return gmmktime( (int) $time_chunks[1], (int) $time_chunks[2], (int) $time_chunks[3], (int) $date_chunks[1], (int) $date_chunks[2], (int) $date_chunks[0] );
+}
```

Comment:
@milanhirapra We need to optimize these 2 functions to single and pass the parameters based on start or end week.